CRM/Contribute/Form/ContributionBase.php

```diff
@@ -207,6 +207,15 @@ class CRM_Contribute_Form_ContributionBase extends CRM_Core_Form {
    */
   public $paymentInstrumentID;
 
+  /**
+   * The contribution ID
+   *
+   * @var int
+   *
+   * @fixme this is retrieved as a URL param. I think it is only set when using the contribution page in "invoice" mode
+   */
+  protected $_ccid;
+
```

Comment:
It's used publicly in Contribution_Main::formRule. BUT...

it seems like it's still able to access it - maybe a php thing I'm not aware of. I would assume the below would give an error, but it doesn't seem to:

```php
class A {
  protected $bar;
}
class B extends A {
  function do_something($z) {
    echo $z->bar; // It's accessing bar as a public member, not as $this->bar, but maybe php looks and sees it's a `B` and it knows we're inside `B`?
  }
}
$form = new B();
$form->do_something($form);
```

So anyway, the rest of the PR looks good.

Reply:
Yeah - I see it - adding a public getter would be one option

![Screenshot from 2021-02-01 13-35-12](https://user-images.githubusercontent.com/336308/106403235-65bfbf00-6492-11eb-9924-d78819ebc43b.png)

Reply:
Thanks for spotting the protected/public thing.